### ISSUE TITLE
[feature] Add a new function repo:resource-available

### ISSUE THREAD

**Issue by joewiz** [eXist-db/exist#3904]
**Is your feature request related to a problem? Please describe.**

With https://github.com/eXist-db/exist/pull/3896, calls to `repo:get-resource` now throw an error when the 2nd argument, `$resource`, does not exist in the package (as stated in [the function's documentation](http://www.exist-db.org/exist/apps/fundocs/view.html?uri=http://exist-db.org/xquery/repo&location=java:org.exist.xquery.modules.expathrepo.ExpathPackageModule&details=true#get-resource.2)).

We now need a `repo:resource-available` function to check if a resource exists before we use `repo:get-resource` to get it. The purpose is analogous to `fn:doc-available` for `fn:doc`, and `util:binary-resource-available` for `util:binary-resource`.

**Describe the solution you'd like**

We need a new function that checks if a resource is available and returns `true()` or `false()` accordingly:

```xquery
declare function repo:resource-available($pkgName as xs:string, $resource as xs:string) as xs:boolean
```

**Describe alternatives you've considered**

In the meantime, a try-catch expression will fix affected code, but it's just a short term workaround.

**Additional context**

The motivation is to help in the fix to https://github.com/eXist-db/existdb-packageservice/issues/19.

**Comment by adamretter** [comment]
@joewiz As #3896 is clearly an behaviour API change. Should we revert it until 6.0.0?

**Comment by joewiz** [comment]
@adamretter Yes, that's probably a good idea! I searched elsewhere in the various eXist projects I've cloned to my computer and saw uses of `repo:get-resource` in:

- exist's expathrepo/repair.xql
- shared-resources's apputil.xql
- eXide's deployment.xql
- tei-publisher-app's components-generate.xql

Not all of these will require the same fix I applied to packageservice, but we should probably move to a pattern where we check for the existence of a resource before getting it—and that would likely not happen until after 5.3.0 anyway.
